packages/vscode-extension/src/webview/components/UrlSelect.tsx

```diff
@@ -33,7 +33,8 @@ function UrlSelect({
   const [currentDynamicSegment, setCurrentDynamicSegment] = React.useState<number>(0);
   const [textfieldWidth, setTextfieldWidth] = React.useState<number>(0);
 
-  const dropdownItemsRef = React.useRef<Array<HTMLDivElement>>([]);
+  const dropdownItems: UrlSelectFocusable[] = [];
```

Comment:
I don't like the idea of creating a new object to pass as a prop to children on each render.
Maybe it's fine in this case, since we want to always store the rendered elements here.
I'd rather someone else give their opinion on this.
@kmagiera @filip131311